src/app/routes/cpsAsset/getInitialData/cpsRecomendations/index.js

```diff
@@ -0,0 +1,76 @@
+import path from 'ramda/src/path';
+import deepClone from 'ramda/src/clone';
+import splitAt from 'ramda/src/splitAt';
+import { optimoSubheadline } from '../convertToOptimoBlocks/utils/helpers';
+
+const isCpsParagraph = (block) => {
+  /* In CPS the following blocks are considered paragraphs
+    Image, Audio, Video, Paragraph
+  */
+  const mediaTypes = ['image', 'audio', 'video'];
+  if (mediaTypes.includes(block.type)) return true;
+
+  if (block.type === 'text') {
+    const paragraphBlocks = path(['model', 'blocks'], block).filter(
+      (b) => b.type === 'paragraph',
+    );
+    return !!paragraphBlocks.length;
+  }
+  return false;
+};
+
+const getNthCpsParagraphIndex = (blocks, n) => {
+  let indexCount = 0; // maybe make it the lastt block just incase there's no paragraph blocks
+  let paragraphCount = 0;
+  blocks.some((block, index) => {
+    if (isCpsParagraph(block)) {
+      paragraphCount += 1;
+    }
+    if (paragraphCount === n) {
+      indexCount = index;
+      return true;
+    }
+    indexCount += 1;
+    return false;
+  });
+  return indexCount;
```

Comment:
had a chat with @lizcameron about 2 things
- Where do we want to place this if we don't have up to 7 paragraphs on a page ?
> For now it shouldn't be rendered if we don't have up to 7 paragraphs. More info will be shared on this after chat with datalab team.

- What do we do if the 7th paragraph is a title block (_**title blocks are also paragraphs**_) of the next block eg. _a social embed block having a preceding paragraph block_ ?
> We'll just assume that's an edge case now because of the local story page data and revisit later to test with live data